### PR TITLE
Replace BinaryIO with IO[bytes]

### PR DESCRIPTION
**Pull request by q0w**
## What do these changes do?
Replace `BinaryIO` with `IO[bytes]`

So `mktar_from_dockerfile` should take `fileobj` as `Union[BytesIO, IO[bytes]]` (`IO[bytes]` for `else` case)  and return `IO[bytes]` because there is not any case with returning `IO[str]`. Also `docker.images.build` should also take `fileobj` as `IO[bytes]`
There is also a discussion about `typing.IO`: https://github.com/python/typing/discussions/829, so protocols are useless due to the amount of use.
## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [ ] Add a new news fragment into the `changes` folder
  * name it `<issue_id>.<type>` for example (588.bug)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by asvetlov** [comment]
From my understanding,  python/typing#829 recommends using protocols for function arguments.
Would you modify the PR to follow this suggestion?

File: aiodocker/utils.py [hunk 1]
```diff
@@ -227,12 +217,12 @@ def clean_filters(filters: Mapping = None) -> str:
     return json.dumps(filters)
 
 
-def mktar_from_dockerfile(fileobject: BinaryIO) -> IO:
+def mktar_from_dockerfile(fileobject: Union[BytesIO, IO[bytes]]) -> IO[bytes]:
```
Comment (q0w) [inline aiodocker/utils.py:220]: I am not sure what to use here as `Tarfile.gettarinfo` expects fileobj as `BytesIO`

File: aiodocker/images.py [hunk 2]
```diff
@@ -14,6 +14,10 @@
     overload,
 )
 
+
+if TYPE_CHECKING:
+    from _typeshed import SupportsRead
```
Comment (asvetlov) [inline aiodocker/images.py:19]: Sorry, an import from a private module is prohibited.
Please copy-paste `SupportsRead` protocol implementation here.

File: aiodocker/utils.py (hunk 1 above)
Comment (asvetlov) [reply aiodocker/utils.py:220]: Ok, let's keep `BytesIO` if there is no choice.

I can recommend making a PR for typeshed; they accept corrections easily and quickly if a patch is correct.

**Review by asvetlov — CHANGES_REQUESTED** [review]

**Review by achimnol — APPROVED** [review]
We need to revisit the test suite (#849), so I'm going to merge this first and do follow-ups.

**Pull request merged**